feat: Integrate Jupiter DEX

Blazebrain opened the pull request:
Issue Number (if Applicable): Fixes #

# Description

Please include a summary of the changes and which issue is fixed / feature is added. 

# Pull Request - Checklist  

- [ ] Initial Manual Tests Passed
- [ ] Double check modified code and verify it with the feature/task requirements
- [ ] Format code
- [ ] Look for code duplication
- [ ] Clear naming for variables and methods
- [ ] Manual tests in accessibility mode (TalkBack on Android) passed

Blazebrain committed:
feat: Integrate Jupiter DEX

Blazebrain committed:
Merge branch 'main' of https://github.com/cake-tech/cake_wallet into CW-1244-Implement-Jupiter-Dex

Blazebrain committed:
feat: Enable internal swaps

Blazebrain committed:
Merge branch 'main' of https://github.com/cake-tech/cake_wallet into CW-1244-Implement-Jupiter-Dex

Blazebrain committed:
feat: implement Jupiter swap execution api and enhance trade handling

- Added executeSwap method to handle signed swap transactions via the execute endpoint.
- Updated signAndPrepareJupiterSwapTransaction to include request ID and handle swap execution response.
- Modified trade details to use txId instead of id for Jupiter trades.
- Enhanced error handling for swap execution with user-friendly messages.
- Updated sendviewmodel to manage trade state updates after transaction commitment.

OmarHatem28 commented on lib/view_model/send/send_view_model.dart:
```diff
@@ -682,6 +691,38 @@ abstract class SendViewModelBase extends WalletChangeListenerViewModel with Stor
+            requestId?.isNotEmpty == true &&
+            solana != null) {
+          try {
+            final actualFee = double.tryParse(trade.memo ?? '0.0') ?? 0.0005;
```
why is the fee stored in memo?

Blazebrain committed:
Merge branch 'main' of https://github.com/cake-tech/cake_wallet into CW-1244-Implement-Jupiter-Dex

Blazebrain committed:
fix: null error after successfully swapping

Blazebrain committed:
fix: Finallyyy fixed the annoying tx history issue for solana dex swaps

Blazebrain committed:
Updates

Blazebrain committed:
fix: update inputAddress to use toAddress in JupiterExchangeProvider

Blazebrain committed:
feat: enhance transaction handling and token balance updates

- Cut down tx fetch/update time for transactions update after swapping
- Added TransactionFetchResult class to hold parsed transactions and token mints.
- Added pollForTransaction method to handle transaction polling with exponential backoff.
- Conditionally hide the external send button based on provider type.

Blazebrain committed:
feat: add fees to trade object and handle null case

Blazebrain committed:
feat: add Jupiter referral fee and account configuration

Blazebrain committed:
Merge branch 'main' of https://github.com/cake-tech/cake_wallet into CW-1244-Implement-Jupiter-Dex

Blazebrain committed:
Merge branch 'main' of https://github.com/cake-tech/cake_wallet into CW-1244-Implement-Jupiter-Dex

Blazebrain committed:
Merge branch 'dev' of https://github.com/cake-tech/cake_wallet into CW-1244-Implement-Jupiter-Dex

Blazebrain committed:
Merge branch 'dev' of https://github.com/cake-tech/cake_wallet into CW-1244-Implement-Jupiter-Dex

OmarHatem28 committed:
Merge branch 'dev' into CW-1244-Implement-Jupiter-Dex

OmarHatem28 commented:
@Blazebrain please check the testers comments left on the ticket
